elements/lisk-chain/src/state_store/account_store.ts

```diff
@@ -137,15 +143,31 @@ export class AccountStore {
 		this._updatedKeys.add(primaryValue);
 	}
 
-	public finalize(batch: BatchChain): void {
+	public finalize(batch: BatchChain): StateDiff {
+		const stateDiff = {} as StateDiff;
+
 		for (const account of this._data.values()) {
 			if (this._updatedKeys.has(account.address)) {
-				const encodedAccount = this._dataAccess.encodeAccount(account);
+				const encodedFinalAccount = this._dataAccess.encodeAccount(account);
+				const dbKey = `${DB_KEY_ACCOUNTS_ADDRESS}:${keyString(account.address)}`;
 				batch.put(
-					`${DB_KEY_ACCOUNTS_ADDRESS}:${keyString(account.address)}`,
-					encodedAccount,
+					dbKey,
+					encodedFinalAccount,
 				);
+
+				if (Object.entries(this._initialAccountValue).length) {
+					const encodedInitialAccount = this._dataAccess.encodeAccount(this._initialAccountValue);
```

Comment:
Not specifically related this this PR, but probably it wouldve been better if the storage returns raw Buffer, and decoding was done in this file, so that this `encode` doesnt need to happen